test/integration/content-sidebar/AdditionalTabs.int.test.js

```diff
@@ -0,0 +1,108 @@
+                                                                <path d="M0 0h24v24H0z" fill="none"/>
+                                                            </svg>`;
+
+const blob = new Blob([svg], { type: 'image/svg+xml' });
+// Used to simulate an icon URL coming back from the server
+const url = URL.createObjectURL(blob);
+
+const tabData = [
+    {
+        id: 200,
+        title: '1', // used as tooltip
+        iconUrl: url,
+        callback: () => {},
+        status: 'ADDED',
+    },
+    {
+        id: 201,
+        title: '2',
+        iconUrl: url,
+        callback: () => {},
+        status: 'ADDED',
+    },
+    {
+        id: 202,
+        title: '3',
+        iconUrl: url,
+        callback: () => {},
+        status: 'ADDED',
+    },
+    {
+        id: -1,
+        title: 'More Apps',
+        callback: () => {},
+        status: 'ADDED',
+    },
+];
+
+const getTabs = () => cy.getByTestId('additionaltab');
+
+describe('additional-tabs', () => {
+    const helpers = {
+        load(additionalProps = {}) {
+            cy.visit('/Elements/ContentSidebar', {
+                onBeforeLoad: contentWindow => {
+                    contentWindow.ADDITIONAL_PROPS = additionalProps;
+                },
+            });
+        },
+    };
+
+    it('should display nothing when additional tabs are disabled', () => {
+        helpers.load({ hasAdditionalTabs: false });
+
+        cy.getByTestId('additionaltabplaceholder').should('not.be.visible');
+    });
+
+    it('should display the loading state', () => {
+        helpers.load({ hasAdditionalTabs: true, additionalTabs: [] });
+        cy.getByTestId('additionaltabplaceholder').should('have.length', 5);
+    });
+
+    it('should render the provided icons and tooltips', () => {
+        helpers.load({ hasAdditionalTabs: true, additionalTabs: tabData });
+
+        cy.getByTestId('additionaltabplaceholder').should('not.be.visible');
+
+        getTabs()
+            .eq(0)
+            .trigger('mouseover');
+
+        cy.getTooltip().contains('1');
+
+        getTabs()
+            .eq(2)
+            .trigger('mouseover');
+
+        // The tabs should be rendering in order
+        cy.getTooltip().contains('3');
+
+        getTabs()
+            .eq(3)
+            .trigger('mouseover');
+
+        // Check that the more items tab rendered as intended
+        cy.getTooltip().contains('More Apps');
+    });
+
+    it('should execute the given callback on click', () => {
+        const stub = cy.stub(); // eslint-disable-line cypress/no-assigning-return-values
+        tabData[0].callback = stub;
+        helpers.load({ hasAdditionalTabs: true, additionalTabs: tabData });
+
+        getTabs()
+            .eq(0)
+            .click()
+            .then(() => expect(stub).to.be.calledWith({ id: 200, callbackData: { status: 'ADDED' } }));
+    });
+
+    it('should properly handle a failed icon load', () => {
+        tabData[1].iconUrl = 'foobar.jpg';
```

Comment:
I just noticed we're mutating the test data from test to test. This could cause issues with tests relying on each other.

Reply:
will address, thanks.